precompiles/ArbNativeTokenManager.go

```diff
@@ -31,7 +35,7 @@ func (con ArbNativeTokenManager) MintNativeToken(c ctx, evm mech, amount huge) e
 
 	evm.StateDB.ExpectBalanceMint(amount)
 	evm.StateDB.AddBalance(c.caller, uint256.MustFromBig(amount), tracing.BalanceIncreaseMintNativeToken)
-	return nil
+	return con.NativeTokenMinted(c, evm, c.caller, amount)
```

Comment:
This is more a nitpick: we could also check if events are being properly generated in tests

Reply:
Done.